plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/subplugin/GodotSubPlugin.kt

```diff
@@ -0,0 +1,52 @@
+package godot.gradle.subplugin
+
+import godot.compilerplugin.CompilerPluginConst
+import godot.gradle.GodotExtension
+import godot.gradle.GodotPlugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.compile.AbstractCompile
+import org.jetbrains.kotlin.gradle.dsl.KotlinCommonOptions
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation
+import org.jetbrains.kotlin.gradle.plugin.KotlinGradleSubplugin
+import org.jetbrains.kotlin.gradle.plugin.SubpluginArtifact
+import org.jetbrains.kotlin.gradle.plugin.SubpluginOption
+
+class GodotSubPlugin : KotlinGradleSubplugin<AbstractCompile> {
+    override fun apply(
+        project: Project,
+        kotlinCompile: AbstractCompile,
+        javaCompile: AbstractCompile?,
+        variantData: Any?,
+        androidProjectHandler: Any?,
+        kotlinCompilation: KotlinCompilation<KotlinCommonOptions>?
+    ): List<SubpluginOption> {
+        val extension = project.extensions.getByType(GodotExtension::class.java)
+        return listOf(
+            SubpluginOption(
+                CompilerPluginConst.CommandLineOptionNames.gdnsDirPathOption,
+                extension.gdnsDir.get().asFile.absolutePath
+            ),
+            SubpluginOption(
+                CompilerPluginConst.CommandLineOptionNames.entryDirPathOption,
+                extension.entrySourceDir.get().asFile.absolutePath
+            )
+        )
+    }
+
+    override fun getCompilerPluginId(): String = CompilerPluginConst.compilerPluginId
+
+    override fun isApplicable(project: Project, task: AbstractCompile): Boolean =
+        project.plugins.hasPlugin(GodotPlugin::class.java)
```

Comment:
check `task` if it is a instance of `KotlinNativeCompile`?

Reply:
:+1:

Reply:
But shouldn't it be `NativeCompilation`?

Reply:
Yeah, I keep forgetting the task class name :joy:

Reply:
ah no you were right. it expects a clas inherited from abstractCompile

Reply:
as discussed on discord: sadly not possible